src/installers.ts

```diff
@@ -459,6 +474,123 @@ export const installArchiveOnlyMod: VortexWrappedInstallFunc = (
   return Promise.resolve({ instructions });
 };
 
+export const testForJsonMod: VortexWrappedTestSupportedFunc = (
+  _api: VortexAPI,
+  log: VortexLogFunc,
+  files: string[],
+  gameId: string,
+): Promise<VortexTestResult> => {
+  // Make sure we're able to support this mod.
+  const correctGame = gameId === GAME_ID;
+  log("info", "Checking JSON files for game: ", gameId);
+  if (!correctGame) {
+    // Not in game mode?
+    return Promise.resolve({
+      supported: false,
+      requiredFiles: [],
+    });
+  }
+  let supported: boolean;
+  const filtered = files.filter(
+    (file: string) => path.extname(file).toLowerCase() === JSON_FILE_EXT,
+  );
+  if (filtered.length === 0) {
+    log("info", "No JSON files");
+    return Promise.resolve({
+      supported: false,
+      requiredFiles: [],
+    });
+  }
+
+  // just make sure we don't somehow have a CET mod that got here
+  const cetModJson = filtered.filter((file: string) =>
+    path.dirname(file).toLowerCase().includes(CET_MOD_CANONICAL_PATH_PREFIX),
```

Comment:
Should we check the path, or check that there is an 'init.lua'? If, god forbid, someone ships a mod without proper CET structure but includes a json file, this would not pick it up. While I don't know of any examples of this happening, being a professional programmer has led me to never trust where an end user provides input.